src/command.rs

```diff
@@ -295,23 +301,102 @@ impl CargoCmd {
     ///
     /// - `cargo 3ds build` and other "build" commands will use their callbacks to build the final `.3dsx` file and link it.
     /// - `cargo 3ds new` and other generic commands will use their callbacks to make 3ds-specific changes to the environment.
-    pub fn run_callback(&self, messages: &[Message]) {
+    pub fn run_callbacks(&self, messages: &[Message], metadata: &Option<Metadata>) {
+        let max_artifact_count = if let Some(metadata) = metadata {
+            metadata.packages.iter().map(|pkg| pkg.targets.len()).sum()
+        } else {
+            0
+        };
+
+        let mut configs = Vec::with_capacity(max_artifact_count);
```

Comment:
I just realized, this `with_capacity` is probably useless since I just assign `configs` directly in the case where we are building stuff. `configs.extend()` is probably more appropriate now that it's got a capacity